tests/test_chemical_parser.py

```diff
@@ -0,0 +1,114 @@
+from src.gudrun_classes.composition import Component
+from unittest import TestCase
+
+
+class TestChemicalParser(TestCase):
+
+    def testValidParse(self):
+
+        formula = "H2O"
+        component = Component(formula)
+        component.parse()
+
+        self.assertEqual(len(component.elements), 2)
+
+        self.assertEqual(component.elements[0].atomicSymbol, "H")
+        self.assertEqual(component.elements[0].massNo, 0)
+        self.assertEqual(component.elements[0].abundance, 2.0)
+
+        self.assertEqual(component.elements[1].atomicSymbol, "O")
+        self.assertEqual(component.elements[1].massNo, 0)
+        self.assertEqual(component.elements[1].abundance, 1.0)
+
+    def testInvalidParse(self):
+
+        formula = "h2o"
+        component = Component(formula)
+        component.parse()
+
+        self.assertEqual(len(component.elements), 0)
+
+    def testD2OParse(self):
+
+        formula = "D2O"
+        component = Component(formula)
+        component.parse()
+
+        self.assertEqual(len(component.elements), 2)
+
+        self.assertEqual(component.elements[0].atomicSymbol, "H")
+        self.assertEqual(component.elements[0].massNo, 0)
```

Comment:
Once earlier bug is fixed, this will be:
```suggestion
        self.assertEqual(component.elements[0].massNo, 2)
```